<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Exception</title>
    <script>
        // aler("aaa") //코드를 잘못 써서 에러. 에러가 발생하면 다음 줄은 진행되지 않음
        try{
            aler("aaa")
        } catch(e){
            document.write(e + "<br>")
        }
        document.write("예외처리가 안되면 안나오는 메시지 <br>")
        document.write("<hr>")

        //finally
        //네트워크 작업에서 예외 발생시에도 Stream을 닫아줘야 할 때 사용 
        try{
            funA()
        } catch(e){
            document.write(e + "<br>")
        } finally{
            document.write("난 무조건 실행돼 <br>")
        }
        document.write("<hr>")


        //throw - 에러는 아니지만 에러로 정하고 싶을 때 사용
        try{
            var a = 10
            var b = 15
            var c = a - b
            if(c < 0) throw "음수 결과는 에러야!"
            document.write("result: " + c + "<br>")
        } catch(e){
            document.write(e + "<br>")
        }
        document.write("<hr>")
        
        window.document.write("aaa")
        window.parseInt()
        window.Array()
    </script>
    
</head>
<body>
    
</body>
</html>